rust/tools/authorization-logic/src/ast.rs

```diff
@@ -72,10 +79,26 @@ pub enum AstFact {
     AstCanSayFact { p: AstPrincipal, f: Box<AstFact> },
 }
 
+#[derive(Copy, Debug, Clone, Serialize, Deserialize)]
+pub enum AstOperator {
+    LessThan,
+    GreaterThan,
+    Equals,
+    NotEquals,
+    LessOrEquals,
+    GreaterOrEquals
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum AstRValue {
+    FlatFactRValue { flat_fact: AstFlatFact},
```

Comment:
What is a `FlatFact`?

Reply:
FlatFact has been here a while. This is the part of the grammar of facts that can't be nested --  they do not contain `canSay` which is the part that can be nested. These need to be separated at the grammar level because allowing nesting on the RHS would make the language undecidable (and untranslatable to datalog).